### ISSUE TITLE
Ubuntu 24.04 (noble) creates an error 404

### ISSUE THREAD

**Issue by Imperial929** [fujiapple852/trippy#1308]
**Describe the bug**
Ign:5 https://ppa.launchpadcontent.net/fujiapple/trippy/ubuntu noble InRelease
Err:6 https://ppa.launchpadcontent.net/fujiapple/trippy/ubuntu noble Release
  404  Not Found [IP: 2620:2d:4000:1::81 443]
E: The repository 'https://ppa.launchpadcontent.net/fujiapple/trippy/ubuntu noble Release' does not have a Release file.

**To Reproduce**
Install trippy on a fresh Ubuntu 24.04 system

**Expected behavior**
Successfull installation (just as with Ubuntu 22.04 LTS

**Environment Info**
Ubuntu 24.04.01 LTS

**Comment by fujiapple852** [comment]
Thank you for flagging this @Imperial929.  

At the time of release of Trippy 0.11.0 (11th Aug 2024), the latest Ubuntu LTS was `jammy`.  It seems that since then a newer Ubuntu LTS (`noble`) has been [released](https://wiki.ubuntu.com/Releases) so it makes sense to release for that as well.

**Comment by fujiapple852** [comment]
I'm having [trouble](https://trippy.zulipchat.com/#narrow/stream/436136-general/topic/Issue.3A.20.20Ubuntu.2024.2E04.20.28noble.29.20creates.20an.20error.20404.20.231308/near/468924376) uploading the same version for `noble`.

TL;DR; when I try to upload Trippy 0.11.0 for noble I get an error that `trippy_0.11.0.orig.tar.gz` already exists as the filename does not contain the suffix `-ppa1~ubuntu24.04` and so clashes with the prior upload for `jammy`.  I don't know how to convince `debuild` to use a different name.

**Comment by fujiapple852** [comment]
I found this: https://help.launchpad.net/Packaging/UploadErrors

**Comment by fujiapple852** [comment]
@Imperial929 this is now resolved, Trippy 0.11.0 has been released for Ubuntu 24.04 (noble).  This is [trippy - 0.11.0-ppa2~ubuntu24.04](https://launchpad.net/~fujiapple/+archive/ubuntu/trippy/+sourcepub/16487661/+listing-archive-extra).

For my own reference, there were two issues:

- The original tarball (`trippy_0.11.0.orig.tar.gz`) is not versioned with the Ubuntu version and so to use it for `noble` I had to download the exact tarball that was previously uploaded for `jammy` and use that (see the helpful comment from the launchpad docs immediately above).
- The (freshly rebuilt) `vendor.tar.zx` was no longer compatible with the (old) `orig` tarball, specifically as the locked version of `arrayvec` (and likely other) crates was different between the src tarball and the vendored crates and so the build [failed](https://launchpadlibrarian.net/748838081/buildlog_ubuntu-noble-amd64.trippy_0.11.0-ppa1~ubuntu24.04_BUILDING.txt.gz).  The solution was the similar to the above, download the existing `trippy_0.11.0-ppa1~ubuntu24.04.debian.tar.xz` file and extract the `vendor.tar.zx` file and use that to build the package.  Note that this required bumping the `REVISION` to 2 such that we upload `trippy_0.11.0-ppa2~ubuntu24.04.debian.tar.xz`.

For the `0.12.0` release I will upload for both `jammy` and `noble` at the same time and so the issues above should not arise.

Aside: there is an in-flight [ITP](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1081023) to package Trippy for upstream Debian, so hopefully PPAs will not be required in the future.
